tests/MockTest/CheckoutTest.php

```diff
@@ -0,0 +1,281 @@
+<?php
+
+namespace Adyen\MockTest;
+
+class CheckoutTest extends TestCaseMock
+{
+    const NO_CHECKOUT_KEY = "Please insert a valid Checkout API Key in your test.ini file";
+    const MISSING_FIELDS = "Missing the following fields:";
+
+    /**
+     * @param $jsonFile Json file location
+     * @param $httpStatus expected http status code
+     *
+     * @dataProvider successPaymentMethodsProvider
+     */
+    public function testPaymentMethodsSuccess($jsonFile, $httpStatus)
+    {
+        // create Checkout client
+        $client = $this->createCheckoutMockClient($jsonFile, $httpStatus);
+
+        // initialize service
+        $service = new \Adyen\Service\Checkout($client);
+
+        $params = array('merchantAccount' => "YourMerchantAccount");
+        $result = $service->paymentMethods($params);
+
+        $this->assertArrayHasKey('paymentMethods', $result);
+
+    }
+
+    public static function successPaymentMethodsProvider()
+    {
+        return array(
+            array('tests/Resources/Checkout/payment-methods-success.json', 200),
+        );
+    }
+
+    /**
+     * @param $jsonFile
+     * @param $httpStatus
+     * @param $expectedExceptionMessage
+     * @dataProvider failurePaymentMethodsProvider
+     *
+     */
+    public function testPaymentMethodsFailure($jsonFile, $httpStatus, $expectedExceptionMessage)
+    {
+        // create Checkout client
+        $client = $this->createCheckoutMockClient($jsonFile, $httpStatus);
+
+        if ($expectedExceptionMessage == self::NO_CHECKOUT_KEY) {
+            $client->setXApiKey("");
+        }
+        // initialize service
+        $service = new \Adyen\Service\Checkout($client);
+
+        $params = array('merchantAccount' => "YourMerchantAccount");
+        try {
+            $result = $service->paymentMethods($params);
+        } catch (\Exception $e) {
+        }
+
+        $this->assertEquals('Adyen\AdyenException', get_class($e));
```

Comment:
better put these exception assertions within catch and add a fail() call after line 58